Rubberduck.Inspections/Concrete/IntegerDataTypeInspection.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Rubberduck.Common;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.Inspections.Resources;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.UI;
+
+namespace Rubberduck.Inspections.Concrete
+{
+    public sealed class IntegerDataTypeInspection : InspectionBase
+    {
+        public IntegerDataTypeInspection(RubberduckParserState state) : base(state, CodeInspectionSeverity.Hint)
+        {
+        }
+
+        public override CodeInspectionType InspectionType { get; } = CodeInspectionType.CodeQualityIssues;
+
+        public override IEnumerable<IInspectionResult> GetInspectionResults()
+        {
+            var interfaceImplementationMembers = UserDeclarations.FindInterfaceImplementationMembers().ToHashSet();
+
+            var excludeParameterMembers = State.DeclarationFinder.FindEventHandlers().ToHashSet();
+            excludeParameterMembers.UnionWith(interfaceImplementationMembers);
+
+            var result = UserDeclarations
+                .Where(declaration =>
+                    declaration.AsTypeName == Tokens.Integer &&
+                    !interfaceImplementationMembers.Contains(declaration) &&
```

Comment:
a little explanation for the exceptions would be appreciated, since it serves as additional documentation